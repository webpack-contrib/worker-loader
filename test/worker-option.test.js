import {
  compile,
  getCompiler,
  getErrors,
  getModuleSource,
  getResultFromBrowser,
  getWarnings,
} from "./helpers";

describe('"workerType" option', () => {
  it('should use "Worker" by default', async () => {
    const compiler = getCompiler("./basic/entry.js");
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource("./basic/worker.js", stats)).toMatchSnapshot(
      "module"
    );
    expect(result).toMatchSnapshot("result");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should support the "Worker" string value', async () => {
    const compiler = getCompiler("./basic/entry.js", { worker: "Worker" });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource("./basic/worker.js", stats)).toMatchSnapshot(
      "module"
    );
    expect(result).toMatchSnapshot("result");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should support the "Worker" object value', async () => {
    const compiler = getCompiler("./basic/entry.js", {
      worker: {
        type: "Worker",
        options: {
          type: "classic",
          name: "worker-name",
        },
      },
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource("./basic/worker.js", stats)).toMatchSnapshot(
      "module"
    );
    expect(result).toMatchSnapshot("result");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should support the "Worker" object value for inline workers with fallback', async () => {
    const compiler = getCompiler("./basic/entry.js", {
      inline: "fallback",
      worker: {
        type: "Worker",
        options: {
          type: "classic",
          name: "worker-name",
        },
      },
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    const moduleSource = getModuleSource("./basic/worker.js", stats);

    expect(moduleSource.indexOf("inline.js") > 0).toBe(true);
    expect(
      moduleSource.indexOf('__webpack_public_path__ + "test.worker.js"') > 0
    ).toBe(true);
    expect(stats.compilation.assets["test.worker.js"]).toBeDefined();
    expect(result).toMatchSnapshot("result");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should support the "Worker" object value for inline workers without fallback', async () => {
    const compiler = getCompiler("./basic/entry.js", {
      inline: "no-fallback",
      worker: {
        type: "Worker",
        options: {
          type: "classic",
          name: "worker-name",
        },
      },
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    const moduleSource = getModuleSource("./basic/worker.js", stats);

    expect(moduleSource.indexOf("inline.js") > 0).toBe(true);
    expect(
      moduleSource.indexOf('__webpack_public_path__ + "test.worker.js"') === -1
    ).toBe(true);
    expect(stats.compilation.assets["test.worker.js"]).toBeUndefined();
    expect(result).toMatchSnapshot("result");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });
});
