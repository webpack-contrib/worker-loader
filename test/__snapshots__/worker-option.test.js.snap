// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"workerType" option should support the "Worker" object value for inline workers with fallback: errors 1`] = `Array []`;

exports[`"workerType" option should support the "Worker" object value for inline workers with fallback: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"workerType" option should support the "Worker" object value for inline workers with fallback: warnings 1`] = `Array []`;

exports[`"workerType" option should support the "Worker" object value for inline workers without fallback: errors 1`] = `Array []`;

exports[`"workerType" option should support the "Worker" object value for inline workers without fallback: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"workerType" option should support the "Worker" object value for inline workers without fallback: warnings 1`] = `Array []`;

exports[`"workerType" option should support the "Worker" object value: errors 1`] = `Array []`;

exports[`"workerType" option should support the "Worker" object value: module 1`] = `
"export default function Worker_fn() {
  return new Worker(__webpack_public_path__ + \\"test.worker.js\\", {\\"type\\":\\"classic\\",\\"name\\":\\"worker-name\\"});
}
"
`;

exports[`"workerType" option should support the "Worker" object value: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"workerType" option should support the "Worker" object value: warnings 1`] = `Array []`;

exports[`"workerType" option should support the "Worker" string value: errors 1`] = `Array []`;

exports[`"workerType" option should support the "Worker" string value: module 1`] = `
"export default function Worker_fn() {
  return new Worker(__webpack_public_path__ + \\"test.worker.js\\");
}
"
`;

exports[`"workerType" option should support the "Worker" string value: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"workerType" option should support the "Worker" string value: warnings 1`] = `Array []`;

exports[`"workerType" option should use "Worker" by default: errors 1`] = `Array []`;

exports[`"workerType" option should use "Worker" by default: module 1`] = `
"export default function Worker_fn() {
  return new Worker(__webpack_public_path__ + \\"test.worker.js\\");
}
"
`;

exports[`"workerType" option should use "Worker" by default: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"workerType" option should use "Worker" by default: warnings 1`] = `Array []`;
